File: 02 - Even Fibonacci Numbers/cpp/main.cpp
#include <iostream>

int fiboEvenSum(int limit)
{
    int x = 1;
    int y = 2;

    int sum = y;

    while (y <= limit)
    {
        int buf = x + y;
        x = y;
        y = buf;

        if (y % 2 == 0)
        {
            sum += y;
        }
    }

    return sum;
}

int main()
{
    int result = fiboEvenSum(4000000);
    std::cout << "result: " << result << std::endl;
}

File: 02 - Even Fibonacci Numbers/nim/main.nim
proc fibo_even_sum(limit: int): int =
    var x: int = 1
    var y: int = 2
    var sum: int = y;

    while y < limit:
        let buf: int = x + y
        x = y
        y = buf

        if y mod 2 == 0:
            sum += y

    sum

let result: int = fibo_even_sum(4000000)
echo "result: ", result
